src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -4754,6 +4754,14 @@ private unsafe char[] GetCanonicalPath(char[] dest, ref int pos, UriFormat forma
                         end += (_info.Offset.Query - _info.Offset.Path);
                     }
                 }
+
+                // On Unix, escape '\\' in path of file uris to '%5C' canonical form.
+                if (!IsWindowsSystem && _syntax.NotAny(UriSyntaxFlags.ConvertPathSlashes) && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && !IsImplicitFile)
```

Comment:
Thanks @Priya91 but we're extra (maybe exaggeratedly so) careful with Uri :).

We'd prefer if the code doesn't exist at all in Windows builds to ensure app-compat in the case where the Uri has been serialized and stored to disk in any form.

You could wrap it with an `#if` or simply move the code to `Uri.Unix.cs`.